src/runway.rs

```diff
@@ -440,10 +490,77 @@ where
         self.send_consensus_notification(NotificationIn::UnitParents(u_hash, p_hashes));
     }
 
+    fn on_newest_response(
+        &mut self,
+        unchecked_response: UncheckedSigned<NewestUnitResponse<H, D, MK::Signature>, MK::Signature>,
+    ) {
+        let starting_round_sender = match self.starting_round_sender.take() {
+            Some(sender) => sender,
+            None => {
+                log::debug!(target: "AlephBFT-member", "Starting round already sent, ignoring newest unit response");
+                return;
+            }
+        };
+        let response = match unchecked_response.check(self.keybox) {
+            Ok(checked) => checked.into_signable(),
+            Err(e) => {
+                log::debug!(target: "AlephBFT-member", "incorrectly signed response: {:?}", e);
+                return;
+            }
+        };
+
+        if response.salt != self.salt {
+            debug!(target: "AlephBFT-member", "Ignoring newest unit response with an unknown salt: {:?}", response);
+            return;
+        }
+
+        if let Some(unchecked_unit) = response.unit {
+            if unchecked_unit.as_signable().creator() != self.index() {
+                log::debug!(target: "AlephBFT-member", "Not our unit in a response:  {:?}", unchecked_unit.into_signable());
+                return;
+            }
```

Comment:
I would suggest running `validate_unit` on the received unit to avoid some strange corner cases. For instance now it seems we forgot to check is the `session_id` is correct.